extensions/all-extensions/pom.xml

```diff
@@ -23,6 +23,25 @@
             <artifactId>mule-module-file</artifactId>
             <version>${project.version}</version>
             <type>zip</type>
+            <exclusions>
```

Comment:
Yes, you are right. We have to do all of that. Don't merge this yet while
I'm trying to add dependency:analyze as part of extensions pom.xml. I'll
let you know when I'm done.

On Thu, Jun 2, 2016 at 3:04 PM, Mariano Gonzalez <notifications@github.com>
wrote:

> In extensions/all-extensions/pom.xml
> <https://github.com/mulesoft/mule/pull/2743#discussion_r65589012>:
>
> > @@ -23,6 +23,25 @@
> >              <artifactId>mule-module-file</artifactId>
> >              <version>${project.version}</version>
> >              <type>zip</type>
> > +            <exclusions>
>
> All of that is fine, but what I'm saying is that the problem goes way
> beyond extensions. It will be very hard for a developer to figure out that
> the commons-io classes that the IDE and test cases allows to use, will not
> be visible in runtime.
>
> What I'm saying is that we should:
>
>    - Make sure that just because my maven module depend on mule-core, it
>    doesn't get exposed to mule-core dependencies which are not going to be
>    available in runtime
>    - Improve our unit test infrastructure so that its classloaders are
>    closer to the runtime ones
>    - Develop a Mojo which automatically adds the exclusions which you're
>    explicitly adding here... Otherwise this doesn't scale
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mulesoft/mule/pull/2743/files/c98fb624efa9b9914f2b46038515d8dd47bd2159#r65589012>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AG96KYRmKhy7a-y283MUzc9pqxC8hbalks5qHxtGgaJpZM4Isq0Y>
> .
>



-- 
<http://www.mulesoft.com/>
Guillermo Fernandes, Senior Software Engineer

Av. Corrientes 316 Entre Piso, Buenos Aires, C1043AAQ, Argentina


<https://www.linkedin.com/company/www.mulesoft.com>
<https://twitter.com/mulesoft>   <https://www.facebook.com/MuleSoft>
<https://www.youtube.com/user/mulesoftvids> We're hiring!
<http://www.mulesoft.com/careers>